packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -165,7 +153,7 @@ export function useLocationDetailView(
 
   const pageItems = React.useMemo(() => {
     const [start, end] = range;
-    return items.slice(start, end);
+    return Array.isArray(items) ? items.slice(start, end) : [];
   }, [range, items]);
```

Comment:
`usePaginate` needs to return `pageItems`, currently we have replicated this code in at least three places

Reply:
Yep, you're totally right. I've moved the process of creating `pageItems` into `usePaginate` as you've suggested